### PR TITLE
8252530: Fix inconsistencies in hotspot whitebox

### PR DESCRIPTION
**Pull request by marschall**
Replace the runtime check and remove the "InVmBuild" suffix as described in the bug.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8252530](https://bugs.openjdk.java.net/browse/JDK-8252530): Fix inconsistencies in hotspot whitebox

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Alex Menkov](https://openjdk.java.net/census#amenkov) (@alexmenkov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3924/head:pull/3924` \
`$ git checkout pull/3924`

Update a local copy of the PR: \
`$ git checkout pull/3924` \
`$ git pull https://git.openjdk.java.net/jdk pull/3924/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3924`

View PR using the GUI difftool: \
`$ git pr show -t 3924`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3924.diff">https://git.openjdk.java.net/jdk/pull/3924.diff</a>

</details>

### CODE REVIEW

**Review by dholmes-ora — APPROVED** [review]
Hi Philippe,

This all seems fine to me!

Thanks,
David

**Comment by dholmes-ora** [comment]
@marschall hotspot changes require at least two reviews (though only 1 official Reviewer) unless deemed to be trivial. So I will wait for a second review before sponsoring.

**Review by alexmenkov — APPROVED** [review]

**Pull request closed without merging**
